src/components/tx/SignOrExecuteForm/PermissionsCheck.tsx

```diff
@@ -0,0 +1,473 @@
+import { dispatchModuleTxExecution } from '@/services/tx/tx-sender'
+import useOnboard from '@/hooks/wallets/useOnboard'
+import { assertOnboard, assertWallet } from '@/utils/helpers'
+import { TxModalContext } from '@/components/tx-flow'
+import { getModuleTransactionId } from '@/services/transactions'
+
+const Role = ({ children }: { children: string }) => {
+  let humanReadableRoleKey = children
+  try {
+    humanReadableRoleKey = decodeBytes32String(children)
+  } catch (e) {}
+
+  return <Chip label={humanReadableRoleKey} />
+}
+
+const PermissionsCheck: React.FC<{ onSubmit?: SubmitCallback; safeTx: SafeTransaction; safeTxError?: Error }> = ({
+  onSubmit,
+  safeTx,
+  safeTxError,
+}) => {
+  const currentChain = useCurrentChain()
+  const onboard = useOnboard()
+  const wallet = useWallet()
+  const { safe } = useSafeInfo()
+
+  const chainId = currentChain?.chainId || '1'
+
+  const [isPending, setIsPending] = useState<boolean>(false)
+  const [isRejectedByUser, setIsRejectedByUser] = useState<boolean>(false)
+  const [submitError, setSubmitError] = useState<Error | undefined>()
+
+  const { setTxFlow } = useContext(TxModalContext)
+
+  const roles = useRoles(safeTx?.data)
+  const allowingRole = roles.find((role) => role.status === Status.Ok)
+
+  // If a user has multiple roles, we should prioritize the one that allows the transaction's to address (and function selector)
+  const mostLikelyRole =
+    allowingRole ||
+    roles.find((role) => role.status !== Status.TargetAddressNotAllowed && role.status !== Status.FunctionNotAllowed) ||
+    roles.find((role) => role.status !== Status.TargetAddressNotAllowed) ||
+    roles[0]
+
+  // Wrap call routing it through the Roles mod with the allowing role
+  const txThroughRole = useExecuteThroughRole({
+    modAddress: allowingRole?.modAddress,
+    roleKey: allowingRole?.roleKey,
+    metaTx: safeTx?.data,
+  })
+  // Estimate gas limit
+  const { gasLimit, gasLimitError } = useGasLimit(txThroughRole)
+  const [advancedParams, setAdvancedParams] = useAdvancedParams(gasLimit)
+
+  const handleExecute = async () => {
+    assertWallet(wallet)
+    assertOnboard(onboard)
+
+    setIsRejectedByUser(false)
+    setIsPending(true)
+    setSubmitError(undefined)
+    setIsRejectedByUser(false)
+
+    if (!txThroughRole) {
+      throw new Error('Execution through role is not possible')
+    }
+
+    const txOptions = getTxOptions(advancedParams, currentChain)
+
+    let txHash: string
+    try {
+      txHash = await dispatchModuleTxExecution({ ...txThroughRole, ...txOptions }, onboard, chainId, safe.address.value)
+    } catch (_err) {
+      const err = asError(_err)
+      if (isWalletRejection(err)) {
+        setIsRejectedByUser(true)
+      } else {
+        trackError(Errors._815, err)
+        setSubmitError(err)
+      }
+      setIsPending(false)
+      return
+    }
+
+    // On success, wait for module tx to be indexed
+    const transactionService = currentChain?.transactionService
```

Comment:
We don't use the transaction service directly. We use the Client Gateway, which has the [module txs endpoint](https://github.com/safe-global/safe-gateway-typescript-sdk/blob/main/src/index.ts#L103) as well.

Reply:
I checked the client gateway, but it doesn't seem to expose the required `transaction_hash` query param: https://github.com/safe-global/safe-gateway-typescript-sdk/blob/main/src/types/api.ts#L151

Reply:
Or is it just a matter of updating the types in safe-gateway-typescript-sdk?